apps/portals/arkportal/src/pages/DatasetDetailsPage.tsx

```diff
@@ -67,8 +67,20 @@ function DatasetDetailsPage() {
                   const hasVersionNumber = Boolean(
                     context.rowData?.versionNumber,
                   )
+                  // PORTALS-3698: Show files for the latest released version rather than the draft (unless this is the first version)
+                  // This is because the Datasets are from an EntityView rather than a Dataset Collection or Table.
+                  let versionNumberToUse = 1
+                  if (
+                    hasVersionNumber &&
+                    context.rowData?.versionNumber !== undefined
+                  ) {
+                    if (context.rowData?.versionNumber > 1) {
+                      // If this is the first version, we want to show the files for the latest version
+                      versionNumberToUse = context.rowData?.versionNumber - 1
+                    }
+                  }
```

Comment:
If the latest snapshot version is deleted, this will probably result in an API error

Reply:
Good call - what if I use the version history (use the first item)?

Reply:
We do a similar thing in the EntityFinder/Dataset Item editor for cases when the user has to select a version. This works!